src/httpClient.ts

```diff
@@ -151,3 +152,113 @@ export class HttpClientResponse<T> implements IHttpClientResponse {
         this.body = body;
     }
 }
+
+export class HttpApiClient extends HttpClient {
+
+    baseUrl: any;
+
+    constructor(baseUrl: string = undefined) {
+        super();
+        this.baseUrl = baseUrl;
+    }
+
+    request(method: string, url: string, data: any, files: any[], options: IHttpClientRequestOptions) {
+
+        if (typeof method !== "string") {
+            throw new Error("method must be a string");
+        }
+
+        if (["GET", "HEAD", "DELETE", "POST", "PATCH", "PUT"].indexOf(method) < 0) {
+            throw new Error("method can be one of GET, HEAD, DELETE, PATCH, POST, PUT");
+        }
+
+        if (typeof url !== "string") {
+            throw new Error("url must be a string");
+        }
+
+        if (typeof data !== "object" && typeof data !== "undefined") {
+            throw new Error("data must be an object or undefined");
+        }
+
+        if (typeof files !== "object" && typeof files !== "undefined") {
+            throw new Error("files must be an object or undefined");
+        }
+
+        if (typeof options !== "object" && typeof options !== "undefined") {
+            throw new Error("options must be an object or undefined");
+        }
+
+        // Validate method/parameter combinations
+        switch (method) {
+            case "GET":
+                if (data || files) {
+                    throw new Error("data and files cannot be defined on a GET call");
+                }
+                break;
+            case "HEAD":
+                if (data || files) {
+                    throw new Error("data and files cannot be defined on a HEAD call");
+                }
+                break;
+            case "DELETE":
+                if (files) {
+                    throw new Error("files cannot be defined on a DELETE call");
+                }
+                break;
+        }
+
+        options = options || {};
+
+        options.method = method;
+        options.url = url;
+
+        if (data) {
+            options.data = data;
```

Comment:
@wmathes     should I remove the constraint, or use:

     if (data !== undefined) {
            options.data = data;
        }
        if (files !== undefined) {
            options.files = files;
        }